src/basilisp/core.lpy

```diff
@@ -1977,6 +1977,16 @@
          (recur ~expr (inc ~idx))
          ~ret))))
 
+(defmacro aslice
+  "Get a portion of the Python list ``array`` using a
+  Python :external:py:obj:`slice` instance with a single ``stop``
+  value, or a combination of ``start`` and ``stop`` and optional
+  ``step`` values.
```

Comment:
I'm wondering if we should add any documentation about the fact that `nil` is equivalent to omission in Python for any of these values.

Reply:
Updated, I've tried to mention `nil` means `omit` in the next paragraph, but please be free to suggest a better wording.